hazelcast/src/main/java/com/hazelcast/query/Predicates.java

```diff
@@ -465,6 +466,11 @@ public AndPredicate(Predicate... predicates) {
 
         @Override
         public Set<QueryableEntry> filter(QueryContext queryContext) {
+            Set<QueryableEntry> optoResult = tryOptimizedFiltering(queryContext);
+            if (optoResult != null) {
+                return optoResult;
+            }
+
             Set<QueryableEntry> smallestIndexedResult = null;
```

Comment:
that's the way to go; however it's not without consequences - e.g. in the case of an error toString() would display that  BETWEEN predicate was used instead of AND. for now I wanted the least risky change and simplest thing possible demonstrating the effect.